src/Angular.js

```diff
@@ -1031,6 +1031,24 @@ function copy(source, destination, maxDepth) {
 
       case '[object Blob]':
         return new source.constructor([source], {type: source.type});
+      case '[object Map]':
+        // If we're in this case we know the environment supports Map
+        /* eslint-disable no-undef */
+        var copiedMap = new Map();
```

Comment:
We access globals through the `window` in order to support non-browser environments (such as jsdom).